builtin/credential/aws/path_config_client.go

```diff
@@ -27,6 +27,24 @@ func pathConfigClient(b *backend) *framework.Path {
 				Default:     "",
 				Description: "URL to override the default generated endpoint for making AWS EC2 API calls.",
 			},
+
+			"iam_endpoint": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Default:     "",
+				Description: "URL to override the default generated endpoint for making AWS IAM API calls.",
+			},
+
+			"sts_endpoint": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Default:     "",
+				Description: "URL to override the default generated endpoint for making AWS STS API calls.",
+			},
+
+			"iam_auth_header_value": &framework.FieldSchema{
```

Comment:
Why is this called the `auth_header_value` but the header value is `AWSIAM-Server-ID`? The header name makes sense but it'd be nice to have correlation, e.g. `iam_server_id_header_value` or `iam_server_id_value` or so.